core/modules/field/field.module

```diff
@@ -1308,6 +1309,72 @@ function field_get_block_list() {
   return $types;
 }
 
+/**
+ * Helper function to retrieve the most commonly used label for a field.
+ *
+ * @param string $machine_name
+ *   The machine name for the field in question.
+ *
+ * @return string
+ *   The most commonly used machine name.
+ */
+function field_get_label($machine_name) {
```

Comment:
So if only $machine_name is possible, this wouldn't actually help in situations where you knew the bundle but didn't know the label? Seems like that should be provided. e.g. `field_get_label('field_body', 'node', 'post')` would get me the label on the post content type.